### PR TITLE
drivers: ethernet: enc424j600: change mac addr runtime

### PR DESCRIPTION
**Pull request by carlescufi**
Add ability to change mac address at runtime using net mgmt config call.

Supersedes: https://github.com/zephyrproject-rtos/zephyr/pull/73453

### CODE REVIEW

File: drivers/ethernet/eth_enc424j600.c [hunk 1]
```diff
@@ -556,6 +556,41 @@ static enum ethernet_hw_caps enc424j600_get_capabilities(const struct device *de
...
+		tmp = config->mac_address.addr[4] | config->mac_address.addr[5] << 8;
+		enc424j600_write_sfru(dev, ENC424J600_SFR3_MAADR3L, tmp);
+
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+
```
Comment (tomi-font) [inline drivers/ethernet/eth_enc424j600.c:593]: Remove the extra newline?

**Review by tomi-font — COMMENTED** [review]
(tip: `Supersedes`, not `Supercedes` :slightly_smiling_face:)

**Review by pdgendt — DISMISSED** [review]

**Comment by tomi-font** [comment]
FYI @jancoow

**Review by tomi-font — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Comment by carlescufi** [comment]
Note: no need to wait for the 48h period with this PR, since it was already reviewed.

**Pull request merged**